src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,413 @@
+        /// </summary>
+        public PinNumberingScheme DefaultPinNumberingScheme
+        {
+            get
+            {
+                // We have only one scheme
+                return PinNumberingScheme.Logical;
+            }
+        }
+
+        /// <summary>
+        /// Attach to this event to retrieve log messages. The Exception argument may be null if it is only an informational message.
+        /// </summary>
+        public event Action<string, Exception?>? LogMessages;
+
+        /// <summary>
+        /// Searches the given list of com ports for a firmata device.
+        /// </summary>
+        /// <remarks>
+        /// Scanning ports and testing for devices may affect unrelated devices. It is advisable to exclude ports known to contain other hardware from this scan.
+        /// A board won't be found if its port is already open (by the same or a different process).
+        /// </remarks>
+        /// <param name="comPorts">List of com ports. Can be used with <see cref="SerialPort.GetPortNames"/>.</param>
+        /// <param name="baudRates">List of baud rates to test. <see cref="CommonBaudRates"/>.</param>
+        /// <returns>A board, already open and initialized. Null if none was found.</returns>
+        public static ArduinoBoard? FindBoard(IEnumerable<string> comPorts, IEnumerable<int> baudRates)
+        {
+            foreach (var port in comPorts)
+            {
+                foreach (var baud in baudRates)
+                {
+                    ArduinoBoard? b = null;
+                    try
+                    {
+                        b = new ArduinoBoard(port, baud);
+                        b.Initialize();
+                        return b;
+                    }
+                    catch (Exception x) when (x is NotSupportedException || x is TimeoutException || x is IOException || x is UnauthorizedAccessException)
+                    {
+                        b?.Dispose();
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Searches all available com ports for an Arduino device.
+        /// </summary>
+        /// <returns>A board, already open and initialized. Null if none was found.</returns>
+        /// <remarks>
+        /// Scanning serial ports may affect unrelated devices. If there are problems, use the
+        /// <see cref="FindBoard(System.Collections.Generic.IEnumerable{string},System.Collections.Generic.IEnumerable{int})"/> overload excluding ports that shall not be tested.
+        /// </remarks>
+        public static ArduinoBoard? FindBoard()
+        {
+            return FindBoard(SerialPort.GetPortNames(), CommonBaudRates());
+        }
+
+        /// <summary>
+        /// Returns a list of commonly used baud rates.
+        /// </summary>
+        public static List<int> CommonBaudRates()
+        {
+            return new List<int>()
```

Comment:
Just curious, where did we get these from?

Reply:
Knowledge. The ones below 115200 are well known values, above that it's more or less arbitrary.